oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -437,6 +445,46 @@ public void getDefaultCredentials_envAndWellKnownFile_envPrecedence() throws IOE
     testUserProvidesToken(testProvider, transportFactory, accessTokenEnv);
   }
 
+  private class LogHandler extends Handler {
+    LogRecord lastRecord;
+
+    public void publish(LogRecord record) {
+      lastRecord = record;
+    }
+
+    public LogRecord getRecord() {
+      return lastRecord;
+    }
+
+    public void close() {}
+    public void flush() {}
+  }
+
+  @Test
+  public void getDefaultCredentials_wellKnownFile_logsGcloudWarning() throws IOException {
+    Logger logger = Logger.getLogger(DefaultCredentialsProvider.class.getName());
+    LogHandler handler = new LogHandler();
+    logger.addHandler(handler);
+
+    File homeDir = getTempDirectory();
+    File configDir = new File(homeDir, ".config");
+    File cloudConfigDir = new File(configDir, DefaultCredentialsProvider.CLOUDSDK_CONFIG_DIRECTORY);
+    InputStream userStream =
+        UserCredentialsTest.writeUserStream(GCLOUDSDK_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN);
+    File wellKnownFile =
+        new File(cloudConfigDir, DefaultCredentialsProvider.WELL_KNOWN_CREDENTIALS_FILE);
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    testProvider.setProperty("os.name", "linux");
+    testProvider.setProperty("user.home", homeDir.getAbsolutePath());
+    testProvider.addFile(wellKnownFile.getAbsolutePath(), userStream);
+
+    testUserProvidesToken(
+        testProvider, GCLOUDSDK_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN);
+    LogRecord message = handler.getRecord();
+    assertEquals(Level.WARNING, message.getLevel());
```

Comment:
If the log never happens, `message` ends up being null and the test errors with a null pointer exception trying to getLevel().

It might be nicer if the test failed. Could you add `assertNotNull(message);`?

Reply:
Done.